pkg/proxy/ipvs/graceful_termination.go

```diff
@@ -81,6 +81,9 @@ func (q *graceTerminateRSList) remove(rs *listItem) bool {
 }
 
 func (q *graceTerminateRSList) flushList(handler func(rsToDelete *listItem) (bool, error)) bool {
+	q.lock.Lock()
```

Comment:
I think this would result in a deadlock because the call `q.remove(rs)` also acquires this lock.

Reply:
Thank you for pointing that out @andrewsykim 
Code has been updated.